flickit-assessment-data/src/main/java/org/flickit/assessment/data/jpa/kit/kitversion/KitVersionJpaRepository.java

```diff
@@ -17,8 +17,11 @@ public interface KitVersionJpaRepository extends JpaRepository<KitVersionJpaEnti
     @Modifying
     @Query("""
             UPDATE KitVersionJpaEntity k
-            SET k.status = :status
+            SET k.status = :status,
+                k.statusVersion = :statusVersion
             WHERE k.id = :id
         """)
-    void updateStatus(@Param("id") long kitVersionId, @Param("status") int status);
+    void updateStatus(@Param("id") long kitVersionId,
+                      @Param("status") int status,
+                      @Param("statusVersion") Long statusVersion);
```

Comment:
**Suggestion:** Ensure that the updateStatus method in the repository handles null values for statusVersion correctly, as the method signature uses the wrapper class Long which can be null. [possible issue, importance: 5]
```suggestion
    void updateStatus(@Param("id") long kitVersionId, @Param("status") int status, @Param("statusVersion") long statusVersion);
```